robottelo/config/settings.py

```diff
@@ -0,0 +1,365 @@
+    configuration is not defined.
+    """
+
+
+def get_project_root():
+    """Return the path to the Robottelo project root directory.
+
+    :return: A directory path.
+    :rtype: str
+    """
+    return os.path.realpath(os.path.join(
+        os.path.dirname(__file__),
+        os.pardir,
+        os.pardir,
+    ))
+
+
+class INIReader(object):
+    # Helper casters
+    cast_boolean = casts.Boolean()
+    cast_list = casts.List()
+    cast_logging_level = casts.LoggingLevel()
+    cast_tuple = casts.Tuple()
+
+    def __init__(self, path):
+        self.config_parser = ConfigParser()
+        with open(path) as handler:
+            self.config_parser.readfp(handler)
+            if sys.version_info[0] < 3:
+                # ConfigParser.readfp is deprecated on Python3, read_file
+                # replaces it
+                self.config_parser.readfp(handler)
+            else:
+                self.config_parser.read_file(handler)
+
+    def get(self, section, option, default=None, cast=None):
+        """Read an option from a section of a INI file.
+
+        The default value will return if the look up option is not available.
+        The value will be cast using a callable if specified otherwise a string
+        will be returned.
+
+        :param section: Section to look for.
+        :param option: Option to look for.
+        :param default: The value that should be used if the option is not
+            defined.
+        :param cast: If provided the value will be cast using the cast
+            provided.
+
+        """
+        try:
+            value = self.config_parser.get(section, option)
+            if cast is not None:
+                if cast is bool:
+                    value = self.cast_boolean(value)
+                elif cast is list:
+                    value = self.cast_list(value)
+                elif cast is tuple:
+                    value = self.cast_tuple(value)
+                else:
+                    value = cast(value)
+        except (NoSectionError, NoOptionError):
+            value = default
+        return value
+
+    def has_section(self, section):
+        """Check if section is available."""
+        return self.config_parser.has_section(section)
+
+
+class FeatureSettings(object):
+    """Settings related to a feature.
+
+    Create a instance of this class and assign attributes to map to the feature
+    options.
+    """
+
+
+class Settings(object):
+    """Robottelo's settings representation."""
+
+    def __init__(self):
+        self._configured = False
+        self._validation_errors = []
+        self.locale = None
+        self.project = None
+        self.reader = None
+        self.rhel6_repo = None
+        self.rhel7_repo = None
+        self.screenshots_path = None
+        self.server = None
+        self.upstream = None
+        self.verbosity = None
+        self.virtual_display = None
+        self.webdriver = None
+        self.window_manager = None
```

Comment:
I see that you are sorting this list in alphabetical order.  I believe this section above can be sorted with respect to the config order so it is easy to read/debug/reference in future.